HW7_6.16.cpp

```diff
@@ -0,0 +1,47 @@
+/******************************************************************************
+
+                              Online C++ Compiler.
+               Code, Compile, Run and Debug C++ program online.
+Write your code in this editor and press "Run" button to compile and execute it.
+
+*******************************************************************************/
+
+#include <iostream>
+#include <cmath>
+
+using namespace std;
+
+int main()
+{
+    int n, osn=2;
+    scanf("%d",&n);
+    int k = 0;
+    for(int i=1;i<=n;i++)
+    {
+        if(pow(2,i)<=n)
+        {
+            k++;
+        }
+    }
+    k++;
+    printf("%d\n",k);
+    int mas[k];
+    int c = 0;
+    while(n>0)
```

Comment:
it's not effective to transfer bits this way / Use either bit-operation or bitset<>